### DISCUSSION TITLE
When is `require` required?

### DISCUSSION THREAD

**Discussion by kusumotonorio** [factor/factor#2420] [Q&A]
I am trying to change [my vocab](https://github.com/kusumotonorio/multi-dispatch/tree/main/mm9) to consist of multiple files. It mimics [`generic`](https://github.com/factor/factor/tree/master/core/generic), and once I've put it into a single file, I'm trying to rebuild it again to mimic the `generic` directory structure, but it's not working yet.
The fact that it doesn't work is a problem, but I have other questions. I would be grateful if you could help me understand Factor.

At the moment, it seems that the files that can be split do not work unless they are specified with the `require` word from the main file.
```factor
"mm9.standard" require
"mm9.hook" require
"mm9.math" require
```
However, the [`generic.factor`](https://github.com/factor/factor/blob/master/core/generic/generic.factor) does not refer to the lower level files as such. Why does `generic` not refer to ~virtual~ lower level files as such, when my vocabulary requires it?

**Comment by mrjbq7** [comment]
Typical vocabulary hierarchy uses ``USING:`` forms, because some part of the code directly depends on symbols published by those other vocabularies.

In the event that an indirect dependency exists, we would use a ``require`` form with a typical comment to make sure it's not cleaned up by anyone:

```factor
"mm9.math" require ! for mathematical operations
```

What do you mean by virtual files?  Are you talking about ``generic.standard`` and ``generic.single`` and ``generic.math``?

This is likely due to ``syntax`` vocabulary depending on those vocabularies, causing them to be loaded early in the process.

**Reply by kusumotonorio** [reply]
> What do you mean by virtual files? Are you talking about generic.standard and generic.single and generic.math?

Sorry. It's a mistake of "lower level". The Japanese equivalent of "virtual" and "lower level" have the same pronunciation "kasou", so I made mistakes when I typed the translation. I noticed the previous one, but there was another one.

**Reply by kusumotonorio** [reply]
I wanted to split the file into several parts because it was too big to handle, but I haven't succeeded in splitting it in the way I want it to work properly. But I haven't succeeded in splitting the file in a working condition, so I'll leave it as a single file for now.
